### DISCUSSION TITLE
How does mu know it crashed?

### DISCUSSION THREAD

**Discussion by Tinkering-Townsperson** [mu-editor/mu#2195] [Q&A] [answered]
My question: How does the mu editor know it crashed? Is there some sort of library it uses to handle crashes?

**Comment by carlosperate** [comment, marked as answer]
You can use the sys.excepthook Python feature to capture uncaught exceptions:
https://docs.python.org/3/library/sys.html#sys.excepthook

You can see how it was implemented in Mu here:
https://github.com/mu-editor/mu/blob/0596a5251ef4615818fe59837837a82f1430bdd4/mu/app.py#L168-L198

https://github.com/mu-editor/mu/blob/0596a5251ef4615818fe59837837a82f1430bdd4/mu/app.py#L234

**Reply by Tinkering-Townsperson** [reply]
Thank you!
